### PR TITLE
Fix swiftlint integration

### PR DESCRIPTION
**Pull request by kevindelord**
# What this PR does
- Remove all code and integration of `CodeBeat`
- Remove all code and integration of `CodeClimate`
- Updated the README
- Remove the crappy ugly shitty code and replace it with something one can actually read
- Adds lots of comment and documentation for further maintenance
- Parse the config files and insert properly the rule declarations

# What's new

### Configuration variants

From now on, the variant config files (eg. `swiftlint+variant.yml`) will not be simply added/append to the default config but parsed into. This means the rule declarations within the following sections will be properly added to the config:

Sections:
- disabled_rules (rule identifiers to exclude from running)
- excluded (paths to ignore during linting. overridden by `included`) 
- opt_in_rules (optional rules that need to be manually opt-in to be used)
  
This major improvement gives more flexibility to the projects.

This is not a breaking change and the logic should be retro-compatible.

### Project Config Template

When a project does not have a `.project-swiftlint.yml` file the script will now copy the (new) template to the 'parent' repo.

### Disabled Rules

The custom disabled rules are not removed from the config file anymore but commented out.
This way we can keep track of all the rules we have and the (few) ones that are disabled.

This also fixes linting warnings and errors.

# Deprecation

### Script Arguments

The following parameters are now 'soft' deprecated (can be given but will be ignored):
- "--buildconfig"
- "--no-codebeat"
- "--no-codeclimate"

# Be careful...

By merging this PR, the swiftlint integration will be fixed.
Which also means lots of project will get hundreds of warnings and possibly compilation errors.

### CODE REVIEW

**Comment by kevindelord** [comment]
Question @konDeichmann: if it does not exist, would the copy of the `.project-swiftlint.yml` in the main repo break the CI workflow?

I guess this step would happen locally on the developers machine anyway.

File: SwiftLint/project-swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+###
+# .project-frameworks.yml
```
Comment (proth) [inline SwiftLint/project-swiftlint.yml:2]: What is this file for?
```suggestion
# .project-swiftlint.yml
```
?

**Review by proth — APPROVED** [review]

File: SwiftLint/project-swiftlint.yml (hunk 1 above)
Comment (kevindelord) [reply SwiftLint/project-swiftlint.yml:2]: just the name of the file

Comment (proth) [reply SwiftLint/project-swiftlint.yml:2]: But what is the file for?
Is it copied to each the project?
What is supposed to be put in there?

Comment (kevindelord) [reply SwiftLint/project-swiftlint.yml:2]: It's the template for the project specific configuration of swiftlint.
It is copied to the main repo if the file does not exist yet.
As explained in the description of this PR:

> Project Config Template
> When a project does not have a .project-swiftlint.yml file the script will now copy the (new) template to the 'parent' repo.

Comment (proth) [reply SwiftLint/project-swiftlint.yml:2]: Ah. Okay. Now I read the PR description.

**Review by konDeichmann — APPROVED** [review]

**Pull request merged**